java/src/jmri/jmrit/roster/Roster.java

```diff
@@ -321,6 +321,23 @@ public RosterEntry getEntryForId(String id) {
         return null;
     }
 
+     /**
+     * Return a list of RosterEntry which have a 
+     * particular DCC address.
+     *
+     * @param a 
+     * @return an ArrayList of matching entries
+     */
+    public List<RosterEntry> getEntriesByDccAddress(String a) {
+        List<RosterEntry> list = new ArrayList<>(); 
```

Comment:
Steve,

 

Another way to think about this is that any changes in the roster structure
would be handled by the common search code. It also would have options for
caching etc to handle performance issues if needed as another example. Not
using that means any improvements put in that matchingList() code would not
help your addition. Just thinking of the usual reasons for not creating new
wheels.

 

-Ken Cameron, Member JMRI Dev Team

www.jmri.org

www.fingerlakeslivesteamers.org

www.cnymod.com

www.syracusemodelrr.org